### PR TITLE
Fix fetch_op_handle

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/10457

### CODE REVIEW

File: paddle/fluid/framework/details/fetch_op_handle.cc [hunk 1]
```diff
@@ -61,9 +61,14 @@ void FetchOpHandle::RunImpl() {
     auto &scope = scopes[i];
     auto *var =
         scope->FindVar(kLocalExecScopeName)->Get<Scope *>()->FindVar(var_name);
+    if (var == nullptr) {
+      scope->FindVar(var_name);
```
Comment (reyoung) [inline paddle/fluid/framework/details/fetch_op_handle.cc]: This line has no effect

File: paddle/fluid/framework/details/fetch_op_handle.cc [hunk 2]
```diff
@@ -49,7 +49,7 @@ void FetchOpHandle::RunImpl() {
       platform::DeviceContextPool::Instance().Get(platform::CPUPlace());
   for (auto *input : inputs_) {
     auto *var = static_cast<VarHandle *>(input);
-    var->generated_op_->Wait(cpu_ctx);
+    if (var->generated_op_) var->generated_op_->Wait(cpu_ctx);
```
Comment (reyoung) [inline paddle/fluid/framework/details/fetch_op_handle.cc]: if () {
}

File: paddle/fluid/framework/details/fetch_op_handle.cc (hunk 1 above)
Comment (chengduoZH) [reply paddle/fluid/framework/details/fetch_op_handle.cc]: Oh, right, `scope->FindVar(kLocalExecScopeName)->Get<Scope *>()` is the subscope of `scope`.

File: paddle/fluid/framework/details/fetch_op_handle.cc (hunk 2 above)
Comment (chengduoZH) [reply paddle/fluid/framework/details/fetch_op_handle.cc]: Done

**Pull request merged**